### PR TITLE
Fix: Widget: 'value' of null errors

### LINKED ISSUE #6460: [Bug] TypeError: Cannot read property 'value' of null
Sentry Issue: [APPSMITH-115](https://sentry.io/organizations/appsmith/issues/2560288953/?referrer=github_integration)

```
TypeError: Cannot read property 'value' of null
  at predicate (components/designSystems/blueprint/DropdownComponent.tsx:217:21)
  at isOptionSelected (components/designSystems/blueprint/DropdownComponent.tsx:216:27)
  at Object.Yt.renderSingleSelectItem [as itemRenderer] (components/designSystems/blueprint/DropdownComponent.tsx:229:38)
  at u.renderItem (../../../../../src/components/query-list/queryList.tsx:362:31)
  at Array.map (<anonymous>)
...
(45 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by ashit-rath**
#### Steps to reproduce:
1. Drag a dropdown widget
2. Open the property pane
3. For the `options` property add the following code
```
{{[1, 2].map(value => {})}}
```
<br>
This issue was found at multiple places where the property expects an `array of object` values.

- Chart widget
- Select widget (Issue raised with this widget)
- List widget
- Map widget
- Multi-select widget
- Radio Group widget

#### Fixes applied:
- Array validation - returns `default || []` when invalid
- Array Object validation - fixes how object index error message computed
- Added `required` validations to raise error when undefined/null is detected.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested locally
Extended existing test cases.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/6460-dropdown-option-undefined 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.85 **(0.01)** | 36.9 **(-0.02)** | 33.13 **(-0.01)** | 55.42 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 37.72 **(-0.88)** | 36.21 **(0)** | 55.08 **(-0.27)**
 :red_circle: | app/client/src/widgets/ListWidget/ListWidget.tsx | 35.45 **(0)** | 21.02 **(-2.68)** | 28.26 **(0)** | 34.65 **(0)**
 :green_circle: | app/client/src/workers/validations.ts | 80.39 **(0.79)** | 68.34 **(0.52)** | 80.95 **(-0.87)** | 82.76 **(0.43)**</details>

### CODE REVIEW

File: app/client/src/widgets/ChartWidget/propertyConfig.ts [hunk 1]
```diff
@@ -181,6 +181,7 @@ export default [
                 children: {
                   type: ValidationTypes.OBJECT,
                   params: {
+                    required: true,
```
Comment (riodeuno) [inline app/client/src/widgets/ChartWidget/propertyConfig.ts:184]: Have we checked how this effects the chart widget? I mean what happens if I pass an empty value in the property pane, or a `[]` or simply a `[{}]`. All of these should correctly validate.
We can add jest test for this kind of configuration, if we don't already have them

File: app/client/src/widgets/RadioGroupWidget.tsx [hunk 2]
```diff
@@ -25,20 +25,28 @@ class RadioGroupWidget extends BaseWidget<RadioGroupWidgetProps, WidgetState> {
             isTriggerProperty: false,
             validation: {
               type: ValidationTypes.ARRAY,
+              unique: ["value"],
```
Comment (riodeuno) [inline app/client/src/widgets/RadioGroupWidget.tsx:28]: Nice catch!

**Review by riodeuno — COMMENTED** [review]
Code looks good. We need to verify the consequences of the changes by testing the updated validations from the UI. Let's manually try multiple variations as in one of the comments, and see if everything works as expected.

**Review by riodeuno — APPROVED** [review]
Approving for tests. Please feel free to merge when the review comments have been addressed.

File: app/client/src/widgets/ChartWidget/propertyConfig.ts (hunk 1 above)
Comment (ashit-rath) [reply app/client/src/widgets/ChartWidget/propertyConfig.ts:184]: Having the `required` only changes the way `Array<null | undefined>` is validated. If it's not required then `null` and `undefined` are allowed.

Right now `[]` is a valid case, but `[{}]` is not (not due to `required: true`). If there are `allowedKeys` and empty object is present, error is raised; which IMO seems correct as we should enforce user to have the appropriate structure if the array is not empty.

I'll add the test case for `[{}]`.

**Review by riodeuno — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@ashit-rath Working fine on deploy preview, can be merged into release.

**Comment by somangshu** [comment]
@ashit-rath latest release merge required here, Please get an approval post that, and once the tests complete do merge this PR

**Review by somangshu — APPROVED** [review]

**Pull request merged**